diff --git a/modules/user/user.admin.inc b/modules/user/user.admin.inc
index 6ca330b..baa33c7 100644
--- a/modules/user/user.admin.inc
+++ b/modules/user/user.admin.inc
@@ -400,7 +400,7 @@ function user_admin_settings() {
     '#default_value' => variable_get('user_picture_default', ''),
     '#size' => 30,
     '#maxlength' => 255,
-    '#description' => t('URL of picture to display for users with no custom picture selected. Leave blank for none.'),
+    '#description' => t('Filename of the picture to display for users with no custom picture selected. It should be placed in the pictures directory. Leave blank for none.'),
   );
   if (module_exists('image')) {
     $form['personalization']['pictures']['settings']['user_picture_style'] = array(
@@ -663,6 +663,18 @@ function user_admin_settings() {
 }
 
 /**
+ * Form validation handler for the user_admin_settings() form.
+ */
+function user_admin_settings_validate($form, &$form_state) {
+  if (!empty($form_state['values']['user_picture_default'])) {
+    $path = file_build_uri($form_state['values']['user_picture_path'] . '/' . $form_state['values']['user_picture_default']);
+    if (!file_valid_uri($path) || !file_exists(drupal_realpath($path))) {
+      form_set_error('user_picture_default', t('The default user image should be placed inside the %path directory.', array('%path' => $form_state['values']['user_picture_path'])));
+    }
+  }
+}
+
+/**
  * Menu callback: administer permissions.
  *
  * @ingroup forms
diff --git a/modules/user/user.module b/modules/user/user.module
index 29d4793..abb037b 100644
--- a/modules/user/user.module
+++ b/modules/user/user.module
@@ -1480,7 +1480,7 @@ function template_preprocess_user_picture(&$variables) {
       }
     }
     elseif (variable_get('user_picture_default', '')) {
-      $filepath = variable_get('user_picture_default', '');
+      $filepath = file_build_uri(variable_get('user_picture_path', '') . '/' . variable_get('user_picture_default', ''));
     }
     if (isset($filepath)) {
       $alt = t("@user's picture", array('@user' => format_username($account)));
diff --git a/modules/user/user.test b/modules/user/user.test
index e2086d4..508ba3d 100644
--- a/modules/user/user.test
+++ b/modules/user/user.test
@@ -1131,26 +1131,6 @@ class UserPictureTestCase extends DrupalWebTestCase {
   }
 
   /**
-   * Test HTTP schema working with user pictures.
-   */
-  function testExternalPicture() {
-    $this->drupalLogin($this->user);
-    // Set the default picture to an URI with a HTTP schema.
-    $images = $this->drupalGetTestFiles('image');
-    $image = $images[0];
-    $pic_path = file_create_url($image->uri);
-    variable_set('user_picture_default', $pic_path);
-
-    // Check if image is displayed in user's profile page.
-    $this->drupalGet('user');
-
-    // Get the user picture image via xpath.
-    $elements = $this->xpath('//div[@class="user-picture"]/img');
-    $this->assertEqual(count($elements), 1, "There is exactly one user picture on the user's profile page");
-    $this->assertEqual($pic_path, (string) $elements[0]['src'], "User picture source is correct.");
-  }
-
-  /**
    * Tests deletion of user pictures.
    */
   function testDeletePicture() {
